### PR TITLE
Stream details design

### PR DESCRIPTION
**Pull request by dousdogukan**
implement CStreamDetails design page ,
add new colors to tailwind,
add svgs for stream details

### CODE REVIEW

File: src/components/CStreamDetailsStatus/index.tsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import cn from 'classnames';
+interface CStreamDetailsStatusProps {
+  type: 'Active' | 'Expired' | 'Pending';
```
Comment (ceydae) [inline src/components/CStreamDetailsStatus/index.tsx:3]: change the type to 'ongoing' | 'expired' | 'pending'

File: src/components/CStreamDetailsStatus/index.tsx [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+}
+
+const CStreamDetailsStatus = ({ type }: CStreamDetailsStatusProps) => {
+  let statusStyle;
```
Comment (ceydae) [inline src/components/CStreamDetailsStatus/index.tsx:7]: give status style a default value (pending for example, and then remove the last else if)

File: src/containers/StreamDetails/index.tsx [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+import CButton from 'src/components/CButton';
+import CStreamDetailsStatus from 'src/components/CStreamDetailsStatus';
+
+import receiveLogo from '/public/images/receive.svg';
+import copyLogo from '/public/images/whiteCopy.svg';
+import shareLogo from '/public/images/share.svg';
+import withdrawLogo from '/public/images/withdrawSolid.svg';
+
+const StreamDetails = () => {
+  const StreamDetailsTitle = (
```
Comment (ceydae) [inline src/containers/StreamDetails/index.tsx:16]: rename to mainTitle

File: src/containers/StreamDetails/index.tsx [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+      <h1 className="text-2xl text-midnightBlue">Status</h1>
+      <CButton
+        variant="simple"
+        color="outline"
+        content="Withdraw"
+        logo={withdrawLogo}
+        className="!px-3 !py-2 h-[40px]"
+      />
+    </div>
+  );
```
Comment (ceydae) [inline src/containers/StreamDetails/index.tsx:40]: move to StatusCard file

File: src/containers/StreamDetails/index.tsx [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+                    height="48"
+                    viewBox="0 0 202 48"
+                    fill="none"
+                  >
+                    <path
+                      d="M0.824463 10C0.824463 4.47715 5.30162 0 10.8245 0H201.824V48H10.8245C5.30161 48 0.824463 43.5229 0.824463 38V10Z"
+                      fill="white"
+                    />
+                    <path
+                      d="M0.824463 10C0.824463 4.47715 5.30162 0 10.8245 0H201.824V48H10.8245C5.30161 48 0.824463 43.5229 0.824463 38V10Z"
```
Comment (ceydae) [inline src/containers/StreamDetails/index.tsx:83]: move the SVG code to another file called StreamProgress

<StreamProgress progress={0 to 100}

File: src/containers/StreamDetails/index.tsx [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+              className="flex flex-col justify-center items-center w-full h-full px-3 py-4"
+              bgColor="royalBlue"
+              borderColor="rgba(0, 0, 0, 0.10)"
+            >
+              <div className="flex justify-between items-center w-full">
+                <div className="flex justify-center items-center h-12 w-[53px] bg-[#442cd6] text-white text-base px-2 py-2.5 rounded-[9px]">
+                  60%
+                </div>
+
+                <div className="border border-white w-[85%] rounded-xl bg-[#442cd6] pl-[-5px]">
```
Comment (ceydae) [inline src/containers/StreamDetails/index.tsx:70]: move this to another file called BlueCard

File: src/containers/StreamDetails/index.tsx [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+          </div>
+        </section>
+      </CPageCard>
+
+      <div>
+        <div className="w-[580px]">
+          <CPageCard title={summaryTitle} className="px-3 py-4 mb-4 w-full">
+            <ul className="grid gap-2 text-midnightBlue">
+              <li className="flex justify-between w-full overflow-hidden whitespace-nowrap items-center bg-alabaster h-10 px-4 text-sm rounded-[10px]">
+                <span>Total amount</span>
```
Comment (ceydae) [inline src/containers/StreamDetails/index.tsx:137]: Add other fields of the stream according the design

File: src/models.ts [hunk 8]
```diff
@@ -41,3 +41,9 @@ export type CustomError = {
   type: 'error';
   message: string;
 };
+
+export interface IFluxityAPIResponse<T> {
+  status: 'error' | 'success';
+  message: string;
+  result: T;
+}
```
Comment (ceydae) [inline src/models.ts:49]: remove this

**Review by ceydae — CHANGES_REQUESTED** [review]
1. Use mock data from the swagger
2. Add all fields to the summary (create a file called summaryFields.ts)

File: src/components/CSummaryField/index.tsx
Comment (ceydae) [inline src/components/CSummaryField/index.tsx:1]: move this to streamdetails folder

File: src/containers/StreamDetails/StatusCard.tsx
Comment (ceydae) [inline src/containers/StreamDetails/StatusCard.tsx:1]: make 2 different status cards, 1. ReceiverStatusCard 2. SenderStatusCard

File: src/containers/StreamDetails/StreamProgress.tsx [hunk 9]
```diff
@@ -0,0 +1,42 @@
+interface StreamProgressProps {
+  progress: string;
```
Comment (ceydae) [inline src/containers/StreamDetails/StreamProgress.tsx:2]: rename to completionPercentage

File: src/components/CStreamDetailsStatus/index.tsx
Comment (ceydae) [inline src/components/CStreamDetailsStatus/index.tsx:1]: rename CStreamStatusButton

File: src/components/CStreamDetailsStatus/index.tsx [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+
+import capitalizeFirstLetter from 'src/utils/capitalizeFirstLetter';
+
+export type status = 'ongoing' | 'expired' | 'pending';
+interface CStreamDetailsStatusProps {
+  type: status;
+  label: status;
+}
+
+const CStreamDetailsStatus = ({ type, label }: CStreamDetailsStatusProps) => {
```
Comment (ceydae) [inline src/components/CStreamDetailsStatus/index.tsx:11]: remove label

File: src/containers/StreamDetails/index.tsx [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+            <p className="text-base">Total amount streamed</p>
+          </div>
+
+          <BlueCard sender={streamData.sender} flowRate={streamData.rate.toString()} />
+        </section>
+      </CPageCard>
+
+      <div>
+        <SummaryFields />
+        <StatusCard />
```
Comment (ceydae) [inline src/containers/StreamDetails/index.tsx:46]: check address

1. no address = ""
2. address = sender (SenderStatusCard)
3. address = recevier (ReceiverStatusCard)

**Review by ceydae — CHANGES_REQUESTED** [review]

File: src/components/CSummaryField/index.tsx
Comment (dousdogukan) [reply src/components/CSummaryField/index.tsx:1]: This component will also be used in CreateStream summary.

**Review by ceydae — APPROVED** [review]

**Pull request merged**